test/testdata/resolver/requires_ancestor_unsupported_generics.rb

```diff
@@ -26,3 +26,34 @@ module M # error: `Test2::M` can't require generic ancestor `Test2::RA` (unsuppo
     requires_ancestor { RA }
   end
 end
+
+module Test3
+  class RA
+    class << self
+      extend T::Sig
+      extend T::Generic
+
+      Elem = type_member
```

Comment:
Can you add another test that uses `type_template` and `def self.` instead of `class << self` + `type_member`? There are some subtle difference between `class << self; Elem = type_member; end` and `Elem = type_template` and I would like to have a test for both.